src/lib/Form/SubmitHandler.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzPlatformAdminUi\Form;
+
+use EzSystems\EzPlatformAdminUi\Form\Data\OnFailureRedirect;
+use EzSystems\EzPlatformAdminUi\Form\Data\OnSuccessRedirect;
+use EzSystems\EzPlatformAdminUi\Notification\NotificationHandlerInterface;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\RouterInterface;
+use Exception;
+
+class SubmitHandler
+{
+    /** @var NotificationHandlerInterface */
+    protected $notificationHandler;
+
+    /** @var RouterInterface */
+    protected $router;
+
+    /**
+     * @param NotificationHandlerInterface $notificationHandler
+     * @param RouterInterface $router
+     */
+    public function __construct(NotificationHandlerInterface $notificationHandler, RouterInterface $router)
+    {
+        $this->notificationHandler = $notificationHandler;
+        $this->router = $router;
+    }
+
+    /**
+     * Wraps business logic with reusable boilerplate code.
+     *
+     * Handles form errors (NotificationHandler:warning).
+     * Handles business logic exceptions (NotificationHandler:error).
+     * Handles form submit redirection (success, failure) if Data object implements fitting interface.
+     *
+     * @param FormInterface $form
+     * @param callable(mixed):array $handler
+     *
+     * @return null|Response
+     */
+    public function handle(FormInterface $form, callable $handler): ?Response
+    {
+        $data = $form->getData();
+
+        if ($form->isValid()) {
+            try {
+                $result = call_user_func_array($handler, [$data]);
```

Comment:
Could be changed to `$result = $handler($data);`. I think it's more readable

Reply:
@webhdx . Told you :P

Reply:
I'm used to seeing `call_user_func_array()` more frequently. But up to you :P